### PR TITLE
remove Storage.Common namespace

### LINKED ISSUE #8052: Storage: Remove Azure.Storage.Common namespace
API Review Feedback https://apiview.azurewebsites.net/Assemblies/Review/b48340c11dab4e88b3b2bd269bc03661#Azure.Storage.Common

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Fixes #8052

### CODE REVIEW

File: sdk/storage/Azure.Storage.Common/src/Constants.cs [hunk 1]
```diff
@@ -480,8 +480,8 @@ internal static class Xml
 
         internal static class GeoRedundantRead
         {
-            internal const string AlternateHostKey = "Azure.Storage.Common.GeoRedundantReadPolicy.AlternateHostKey";
-            internal const string ResourceNotReplicated = "Azure.Storage.Common.GeoRedundantReadPolicy.ResourceNotReplicated";
+            internal const string AlternateHostKey = "Azure.Storage.GeoRedundantReadPolicy.AlternateHostKey";
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Common/src/Constants.cs]: Nice catch

File: sdk/storage/Azure.Storage.Common/tests/StreamPartitionerTests.cs [hunk 2]
```diff
@@ -133,7 +132,7 @@ public async Task GetNextPartitionAsync_1GB()
                         }
                     }
 
-                    Assert.IsTrue(GC.GetTotalMemory(true) - memoryStart < 8 * Constants.DefaultBufferSize); // TODO Assuming at most 8 buffers allocated
+                    Assert.IsTrue(GC.GetTotalMemory(true) - memoryStart < 8 * Storage.Constants.DefaultBufferSize); // TODO Assuming at most 8 buffers allocated
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Common/tests/StreamPartitionerTests.cs]: Why was this change necessary, out of curiosity?

**Review by tg-msft — APPROVED** [review]

File: sdk/storage/Azure.Storage.Common/tests/StreamPartitionerTests.cs (hunk 2 above)
Comment (JoshLove-msft) [reply sdk/storage/Azure.Storage.Common/tests/StreamPartitionerTests.cs]: There is a different Constants file in Storage.Tests

**Review by amnguye — APPROVED** [review]
